weave/tests/test_uploader.py

```diff
@@ -1347,3 +1348,48 @@ def test_upload_correct_version_number(test_basket):
         manifest_dict = json.load(file)
 
     assert manifest_dict["weave_version"] == weave.__version__
+
+def test_upload_from_s3fs(test_basket):
+    """Test that a basket can be uploaded from s3fs to the local file
+    system or s3fs.
+    """
+    pantry_path = os.path.join(test_basket.pantry_path, "test-pantry-1")
+
+    test_basket.file_system.mkdir(pantry_path)
+
+    tmp_txt_file = test_basket.tmpdir.join("test.txt")
+    tmp_txt_file.write("this is a test")
+
+    nested_path = os.path.join(pantry_path, "text.txt")
+    file_to_move = os.path.join(test_basket.pantry_path, "text.txt")
+
+    if test_basket.file_system == local_fs:
+        minio_path = "test-source-file-system"
+        s3fs.mkdir(minio_path)
+        file_to_move = os.path.join(minio_path, "test.txt")
+
+    # Must upload a file because Minio will remove empty directories
+    if test_basket.file_system == local_fs:
+        s3fs.upload(str(tmp_txt_file.realpath()), file_to_move)
+    else:
+        test_basket.file_system.upload(str(tmp_txt_file.realpath()), file_to_move)
+    test_basket.file_system.upload(str(tmp_txt_file.realpath()), nested_path)
+    # Make the Pantries.
+    pantry_1 = Pantry(
+        IndexPandas,
```

Comment:
Do you think we should we test this with IndexSQLite as well?

Reply:
Good catch. I just added this to the same test.